File: 3451-string-compression-iii/3451-string-compression-iii.py
class Solution:
    def compressedString(self, word: str) -> str:
        # TC- O(n)
        # SC- O(n)
        comp = ""
        count = 1
        n = len(word)
        ch = word[0]

        for i in range(1, n):
            if word[i] == ch and count < 9:
                count += 1
            else:
                comp += str(count) + ch
                ch = word[i]
                count = 1
        comp += str(count) + ch
        return comp
        
        

File: 3451-string-compression-iii/3451-string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        StringBuilder comp = new StringBuilder();
        int count = 1;
        int n = word.length();
        char comparator = word.charAt(0);

        for(int i = 1; i < n; i++) {
            if(word.charAt(i) == comparator && count < 9) {
                count++;
            } else {
                comp.append(count).append(comparator);
                comparator = word.charAt(i);
                count = 1;
            }
        }

        comp.append(count).append(comparator);
        return comp.toString();
        
    }
}